File: Manufacturing/src/CSharp/Nuget/Package/Image.cs
/* Copyright (c) Microsoft Corporation. All rights reserved.
   Licensed under the MIT License. */
namespace Microsoft.Azure.Sphere.DeviceAPI
{
    /// <summary>
    /// Device REST APIs to manage device images.
    /// </summary>
    public static class Image
    {
        /// <summary>Makes a REST "GET" request to retrieve a list of images on an attached device.</summary>
        /// <returns>All images running on the device as a string on success. An exception will be thrown on error.</returns>
        public static string GetImages()
        {
            return RestUtils.GetRequest("images");
        }
    }
}

File: Manufacturing/src/Python/package/azuresphere_device_api/image.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

import azuresphere_device_api.utils as utils

__all__ = ['get_images']


def get_images() -> dict:
    """Makes a REST "GET" request to list the images currently on the attached device.

    :return: All images running on the device on success. An exception will be thrown on error.
    :rtype: dict[str, str]
    :raises: requests.exceptions
    :raises: AzureSphereDeviceApiException
    """
    return utils.get_request("images")
